homeassistant/components/habitica/config_flow.py

```diff
@@ -0,0 +1,107 @@
+"""Config flow for habitica integration."""
+import logging
+
+import requests
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_API_KEY, CONF_NAME, CONF_URL, HTTP_OK
+
+from .const import (  # pylint: disable=unused-import
+    API_STATUS,
+    API_USER_PROFILE,
+    CONF_API_USER,
+    DEFAULT_URL,
+    DOMAIN,
+)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_USER): str,
+        vol.Required(CONF_API_KEY): str,
+        vol.Optional(CONF_NAME): str,
+        vol.Optional(CONF_URL, default=DEFAULT_URL): str,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+
+    def check_api_status(url=DEFAULT_URL) -> bool:
+        request = requests.get(API_STATUS)
```

Comment:
We should use the library here to validate the connection.

Reply:
@MartinHjelmare, I believe that the `habitipy` library does not provide any validation utilities. Is there any other conventional validation library used around the source?

Reply:
No, the idea is that the library should be used. Can't we tell by using some library API calls if the credentials etc worked?

Reply:
Changed to make an api call catching an exception